chain-abci/src/storage/tx.rs

```diff
@@ -178,27 +184,37 @@ pub fn verify(
                     }
                 }
             }
-            // check sum(input amounts) == sum(output amounts)
-            // TODO: do we allow "burn"? i.e. sum(input amounts) >= sum(output amounts)
-            let outsum = maintx.get_output_total();
+            // check sum(input amounts) >= sum(output amounts) + minimum fee
+            // TODO: should the fee be fixed / validation would reject TX if it pays more than the below minimum?
+            let min_fee: Coin = app_state
+                .fee_policy
+                .calculate_fee(txsize)
+                .expect("invalid fee policy")
+                .to_coin();
+
+            let outsum = maintx.get_output_total().and_then(|x| x + min_fee);
             if outsum.is_err() {
                 return Err(Error::InvalidSum(outsum.unwrap_err()));
             }
             let outcoins = outsum.unwrap();
-            if incoins != outcoins {
+            if incoins < outcoins {
                 return Err(Error::InputOutputDoNotMatch);
             }
+            (incoins - outcoins).and_then(|x| x + min_fee).unwrap()
```

Comment:
Here, we are marking all the amount that's left after `(input_coins - output_coins)` as `fee`. I'll wait for @CeruleanAtMonaco to comment on this.

Reply:
I think it's fine to leave it the bitcoin way so that if the network is really congested, higher fees transactions are natural targets to get prioritized.